provider-service/base/pkg/server/resource/experiment.go

```diff
@@ -11,39 +17,52 @@ func (*Experiment) Type() string {
 }
 
 func (e *Experiment) Create(body []byte) (ResponseBody, error) {
+	logger := common.LoggerFromContext(e.Ctx)
 	ed := ExperimentDefinition{}
 	if err := json.Unmarshal(body, &ed); err != nil {
+		logger.Error(err, "Create failed to unmarshal ExperimentDefinition")
 		return ResponseBody{}, &UserError{err}
 	}
 
 	id, err := e.Provider.CreateExperiment(ed)
 	if err != nil {
+		logger.Error(err, "CreateExperiment failed")
 		return ResponseBody{}, err
 	}
+	logger.Info("CreateExperiment succeeded", "response id", id)
 
 	return ResponseBody{
 		Id: id,
 	}, nil
 }
 
 func (e *Experiment) Update(id string, body []byte) (ResponseBody, error) {
+	logger := common.LoggerFromContext(e.Ctx)
 	ed := ExperimentDefinition{}
 	if err := json.Unmarshal(body, &ed); err != nil {
+		logger.Error(err, "Update failed to unmarshal ExperimentDefinition")
```

Comment:
same re-wording here
```suggestion
		logger.Error(err, "Failed to unmarshal ExperimentDefinition while updating Experiment")
```

Reply:
https://github.com/sky-uk/kfp-operator/pull/454/commits/85070452ad75b0e2fa2d23670e8ee4a4cfa9141e